examples/DynamicCantileverCase/dynamic_cantilever.py

```diff
@@ -0,0 +1,148 @@
+from dynamic_cantilever_helper import DynamicCantileverVibration
+
+
+def simulate_dynamic_cantilever_with(
+    density=2000,
+    n_elem=100,
+    final_time=300,
+    mode=0,
+    rendering_fps=30,  # For visualization
+):
+    class DynamicCantileverSimulator(BaseSystemCollection, Constraints, CallBacks):
+        pass
+
+    cantilever_sim = DynamicCantileverSimulator()
+
+    # Add test parameters
+    start = np.zeros((3,))
+    direction = np.array([1.0, 0.0, 0.0])
+    normal = np.array([0.0, 1.0, 0.0])
+    base_length = 1
+    base_radius = 0.02
+    base_area = np.pi * base_radius ** 2
+    youngs_modulus = 1e5
+
+    moi = np.pi / 4 * base_radius ** 4
+
+    dl = base_length / n_elem
+    dt = dl * 0.05
+    step_skips = int(1.0 / (rendering_fps * dt))
+
+    # Add Cosserat rod
+    cantilever = CosseratRod.straight_rod(
+        n_elem,
+        start,
+        direction,
+        normal,
+        base_length,
+        base_radius,
+        density,
+        0.0,
+        youngs_modulus,
+    )
+
+    # Add constraints
+    cantilever_sim.append(cantilever)
+    cantilever_sim.constrain(cantilever).using(
+        OneEndFixedBC, constrained_position_idx=(0,), constrained_director_idx=(0,)
+    )
+
+    end_velocity = 0.005
+    vibration = DynamicCantileverVibration(
+        base_length,
+        base_area,
+        moi,
+        youngs_modulus,
+        density,
+        mode=mode,
+        end_velocity_z=end_velocity,
+    )
+
+    input_velocity = vibration.get_initial_velocity_profile(
+        cantilever.position_collection[0, :]
+    )
+    cantilever.velocity_collection[2, :] = input_velocity
+
+    # Add call backs
+    class CantileverCallBack(CallBackBaseClass):
+        def __init__(self, step_skip: int, callback_params: dict):
+            CallBackBaseClass.__init__(self)
+            self.every = step_skip
+            self.callback_params = callback_params
+
+        def make_callback(self, system, time, current_step: int):
+
+            if current_step % self.every == 0:
+
+                self.callback_params["time"].append(time)
+                self.callback_params["position"].append(
+                    system.position_collection.copy()
+                )
+                self.callback_params["z_position"].append(
+                    system.position_collection[2, -1].copy()
+                )
+                return
+
+    recorded_history = defaultdict(list)
+    cantilever_sim.collect_diagnostics(cantilever).using(
+        CantileverCallBack, step_skip=step_skips, callback_params=recorded_history
+    )
+    cantilever_sim.finalize()
+
+    total_steps = int(final_time / dt)
+    print(f"Total steps: {total_steps}")
+
+    timestepper = PositionVerlet()
+
+    integrate(
+        timestepper,
+        cantilever_sim,
+        final_time,
+        total_steps,
+    )
+
+    # FFT
+    amps = np.abs(fft(recorded_history["z_position"]))
+    f_length = len(amps)
+    amps = amps * 2 / f_length
```

Comment:
Same as above, expand up on the name.

Reply:
[3e1ca62](https://github.com/GazzolaLab/PyElastica/pull/173/commits/3e1ca62f44482de6dc719256fd47975ea3aa5037)